mergify_engine/actions/queue.py

```diff
@@ -394,15 +400,22 @@ async def _should_be_queued(self, ctxt: context.Context, q: queue.QueueT) -> boo
         ]
 
     async def _should_be_merged(
-        self, ctxt: context.Context, rule: "rules.EvaluatedRule", q: queue.QueueT
+        self,
+        ctxt: context.Context,
+        q: queue.QueueBase,
+        p: typing.Optional[QueuePauseData],
     ) -> bool:
-        if not await q.is_first_pull(ctxt):
+        is_first_pull = await q.is_first_pull(ctxt)
+        if not is_first_pull:
             return False
 
         car = typing.cast(merge_train.Train, q).get_car(ctxt)
         if car is None:
             return False
 
+        if p is not None:
+            return False
```

Comment:
Might be worth checking this first?
That'd avoid requesting stuff from what I understand.